src/System.Linq.Expressions/src/System/Linq/Expressions/IndexExpression.cs

```diff
@@ -472,7 +472,7 @@ private static void ValidateAccessorArgumentTypes(MethodInfo method, ParameterIn
 
                     Type pType = pi.ParameterType;
                     if (pType.IsByRef) throw Error.AccessorsCannotHaveByRefArgs($"{nameof(indexes)}[{i}]");
-                    TypeUtils.ValidateType(pType, $"{nameof(indexes)}[{i}]");
+                    TypeUtils.ValidateType(pType, nameof(indexes));
```

Comment:
Any downside to losing the index from the paramName? If the index is useful, maybe `TypeUtils.ValidateType` needs an overload that takes the index as a param (or optional parameter), so the string format allocation only occurs during the error path.